### PR TITLE
18502 Filer conversion filing updates

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#18502

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: queue_services/entity-filer/src/entity_filer/filing_processors/conversion.py [hunk 1]
```diff
@@ -111,6 +111,22 @@ def _process_firms_conversion(
...
+            _update_person_change(
+                legal_entity,
+                filing_rec,
+                conversion_filing,
+                filing_meta
+            )
+
     # Name change if present
     with suppress(IndexError, KeyError, TypeError):
         name_request_json = dpath.util.get(
```
Comment (vysakh-menon-aot) [inline queue_services/entity-filer/src/entity_filer/filing_processors/conversion.py:132]: already covered in each function for sp and gp

Comment (vysakh-menon-aot) [inline queue_services/entity-filer/src/entity_filer/filing_processors/conversion.py:132]: already covered in each function for sp and gp

**Comment by vysakh-menon-aot** [comment]
startDate has to be handled separately in each function

**Comment by vysakh-menon-aot** [comment]
Can you try to call existing function instead of copying it here?

File: queue_services/entity-filer/src/entity_filer/filing_processors/conversion.py [hunk 2]
```diff
@@ -165,6 +181,155 @@ def _process_firms_conversion(
...
+
+
+def _update_person_change(
+        legal_entity: LegalEntity,
+        change_filing_rec: Filing,
+        change_filing: Dict,
+        filing_meta: FilingMeta,
+):
+    name_request = dpath.util.get(change_filing, "/conversion/nameRequest", default=None)
+    identifier = dpath.util.get(change_filing_rec.filing_json, "filing/business/identifier")
```
Comment (vysakh-menon-aot) [inline queue_services/entity-filer/src/entity_filer/filing_processors/conversion.py:250]: @argush3 I am not sure how json going to be. This is the same assumption as we have in changeOfRegistration that we receive identifier of SP (not the person) from alternateName

**Comment by argush3** [comment]
As Vysakh mentioned, there is a lot of duplication in this code. 

Can we update `_update_partner_change` and `_update_person_change` from changeOfRegistration filing processor to be more generic to support conversion filings as well?

And can we just import `get_partnership_name`.

File: queue_services/entity-filer/src/entity_filer/filing_processors/conversion.py (hunk 2 above)
Comment (argush3) [reply queue_services/entity-filer/src/entity_filer/filing_processors/conversion.py:232]: We can leave the same as until we get things sorted out.  Will cut tickets once we know how to move forward.

File: queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/partner.py
Comment (argush3) [inline queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/partner.py:1]: can we rename `partner.py` to `alternate_name.py`?

Comment (kzdev420) [reply queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/partner.py:1]: I renamed.

anyway, it looks good to you, I will complete others too.

Comment (argush3) [reply queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/partner.py:1]: thanks, looks good

File: queue_services/entity-filer/src/entity_filer/filing_processors/change_of_registration.py [hunk 3]
```diff
@@ -86,121 +89,121 @@ def post_process(business: LegalEntity, filing: Filing):
...
-        proprietor.alternate_names.append(new_alternate_name)
-
-        filing_meta.change_of_registration = {
-            **filing_meta.change_of_registration,
-            "fromLegalName": alternate_name.name,
-            "toLegalName": to_legal_name,
-        }
+# def _update_partner_change(
+#         legal_entity: LegalEntity,
+#         change_filing_rec: Filing,
```
Comment (argush3) [inline queue_services/entity-filer/src/entity_filer/filing_processors/change_of_registration.py:94]: can all of the commented out code be removed?

Comment (kzdev420) [reply queue_services/entity-filer/src/entity_filer/filing_processors/change_of_registration.py:94]: Yes, I will do.
But, I just wanted you to confirm my changes before I remove the commented-out code.

File: queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+            )
+        
+        if start := change_filing.get("filing", {}).get(f"{filint_type}", {}).get("startDate"):
+            start_date = LegislationDatetime.as_utc_timezone_from_legislation_date_str(
+                start
+            )
+        elif change_filing.effective_date:
+            start_date = change_filing.effective_date.isoformat()
+        else:
+            start_date = LegislationDatetime.now()
```
Comment (vysakh-menon-aot) [inline queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py:136]: Did you copy this from registration? This is not correct. In change of registration start date cannot be updated, so keep the existing value (`start_date=alternate_name.start_date,`) and in conversion it is optional to update start date. Update this accordingly

Comment (vysakh-menon-aot) [inline queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py:136]:
```suggestion
        else:
            start_date = alternate_name.start_date
```

Comment (kzdev420) [reply queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py]: updated

Comment (kzdev420) [reply queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py]: updated

File: queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+
+        new_alternate_name = AlternateName(
+            bn15=alternate_name.bn15,
+            change_filing_id=change_filing_rec.id,
+            end_date=None,
+            identifier=legal_entity.identifier,
+            name=to_legal_name,
+            name_type=AlternateName.NameType.OPERATING,
+            start_date=change_filing.get("startDate"),
```
[the remark this answers is not in the document]
Comment (kzdev420) [reply queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py]: updated

File: queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+from entity_filer.exceptions import DefaultException
+from entity_filer.filing_meta import FilingMeta
+from entity_filer.filing_processors.filing_components import legal_entity_info
+from entity_filer.filing_processors.filing_components.parties import get_or_create_party
+from entity_filer.utils.legislation_datetime import LegislationDatetime
+
+
+def update_partner_change(
+        legal_entity: LegalEntity,
+        filint_type: str,
```
[the remark this answers is not in the document]
Comment (kzdev420) [reply queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py]: fixed

File: queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+
+
+        new_alternate_name = AlternateName(
+            bn15=alternate_name.bn15,
+            change_filing_id=change_filing_rec.id,
+            end_date=None,
+            identifier=legal_entity.identifier,
+            name=to_legal_name,
+            name_type=AlternateName.NameType.OPERATING,
+            start_date=alternate_name.start_date,
```
Comment (vysakh-menon-aot) [inline queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py:65]: this should be same as `update_proprietor_change`

Comment (kzdev420) [reply queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py]: updated

**Review by argush3 — APPROVED** [review]

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**